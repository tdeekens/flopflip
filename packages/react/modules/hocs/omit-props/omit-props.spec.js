import React from 'react';
import { shallow } from 'enzyme';

import omitProps from './omit-props';

const Component = () => null;
const createTestProps = (custom = {}) => ({
  a: 1,
  b: 2,
  c: 3,

  ...custom,
});

describe('rendering', () => {
  let EnhancedComponent;
  let props;
  let wrapper;

  describe('with multiple props', () => {
    beforeEach(() => {
      EnhancedComponent = omitProps('a', 'b')(Component);
      props = createTestProps();

      wrapper = shallow(<EnhancedComponent {...props} />);
    });

    it('should omit multiple props', () => {
      expect(wrapper.find(Component)).toMatchSnapshot();
    });
  });

  describe('without any props', () => {
    beforeEach(() => {
      EnhancedComponent = omitProps()(Component);
      wrapper = shallow(<EnhancedComponent {...props} />);
    });

    it('should do nothing', () => {
      expect(wrapper.find(Component)).toMatchSnapshot();
    });
  });
});
