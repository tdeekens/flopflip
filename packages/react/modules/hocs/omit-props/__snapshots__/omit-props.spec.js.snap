// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering with multiple props should omit multiple props 1`] = `
<Component
  b={2}
  c={3}
/>
`;

exports[`rendering without any props should do nothing 1`] = `
<Component
  a={1}
  b={2}
  c={3}
/>
`;
